### PR TITLE
Add fee schedule generation tool + base usage and canonical price of TokenFeeScheduleUpdate

### LINKED ISSUE #1735: Implement resource usage estimate for new TokenFeeScheduleUpdate + generate fee schedule entry
### Problem

The new `TokenFeeScheduleUpdate` op needs a resource usage estimator, and an accompanying _feeSchedules.json_ entry that charges the expected price to a "standard" operation (initially `$0.001` to align with the [published price of `TokenUpdate`](https://docs.hedera.com/guides/mainnet/fees)).

The standard operation will be assumed to:
- Use a single signature (i.e. the payer account's key has been assigned the "fee schedule admin" role); and,
- Update a fee schedule, adding a single HTS-denominated fixed fee; and,
- Target a token with a lifetime of 7776000 seconds (i.e. the default auto-renew period).

### Solution

Complete the "Fees" checklist in https://github.com/hashgraph/hedera-services/issues/1722.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
💰 &nbsp;**Fees**
- Closes #1735 
- Translates to Java the [fee schedule generation algorithm](https://github.com/hashgraph/hedera-fee-tool-js/blob/master/src/calculator/Price.js#L221) from the ReactJS app in the _hedera-fee-tool-js_ repo. 
- [x] Create a `TokenOpsUsage` in _hapi-fees/_ that implements `TokenOpsUsage.feeScheduleUpdateUsage()`.
- [x] Extend `AccessorBasedUsages` to support the `TokenFeeScheduleUpdate` function via the above method. 
- [x] Update the `SpanMapManager` to set basic metadata for `TokenFeeScheduleUpdate`.
- [x] Create the `FeeSchedules` class to generate a _feeSchedules.json_ entry for an operation given (1) its desired "base price" in USD; and (2) the `UsageAccumulator` that contains the usage for the base operation.
- [x] Add a _feeSchedule.json_ entry for `TokenFeeScheduleUpdate` computed from a base price of $0.001 a la `TokenUpdate`, and a base usage that adds one HTS-denominated fee to a token with a lifetime of 3 months.

### CODE REVIEW

File: hapi-fees/src/main/java/com/hedera/services/pricing/AssetsLoader.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		try (var fin = AssetsLoader.class.getClassLoader().getResourceAsStream(CONSTANT_WEIGHTS_RESOURCE)) {
+			final var om = new ObjectMapper();
+			final var constWeights = om.readValue(fin, Map.class);
+
+			final Map<HederaFunctionality, BigDecimal> typedConstWeights = new EnumMap<>(HederaFunctionality.class);
+			constWeights.forEach((funcName, weight) -> {
+				final var function = HederaFunctionality.valueOf((String) funcName);
+				final var bdWeight = (weight instanceof Double)
+						? BigDecimal.valueOf((Double) weight)
+						: BigDecimal.valueOf((Integer) weight);
```
Comment (ljianghedera) [inline hapi-fees/src/main/java/com/hedera/services/pricing/AssetsLoader.java:63]: nit: can we require the data source always to be provided as decimal format, so no need to consider the `Integer` case?

File: hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+				SINGLE_SIG_USAGE,
+				NO_MEMO_AND_NO_EXPLICIT_XFERS,
+				opMeta,
+				feeScheduleCtx,
+				into);
+		return into;
+	}
+
+	private UsageAccumulator hbarCryptoTransfer() {
+		throw new AssertionError("Not implemented!");
```
Comment (ljianghedera) [inline hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:136]: Why not provide a implementation for this case? I thought hbar transfer can be considered as the `true` base scenario for CryptoTransfer operation.

Comment (tinker-michaelj) [reply hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:136]: Creating a separate issue to do this once I know Product's requirements for the base price of an NFT transfer. 😄

File: hapi-fees/src/main/java/com/hedera/services/pricing/AssetsLoader.java (hunk 1 above)
Comment (tinker-michaelj) [reply hapi-fees/src/main/java/com/hedera/services/pricing/AssetsLoader.java]: Agreed, no need for this check; removed.

File: hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java (hunk 2 above)
Comment (ljianghedera) [reply hapi-fees/src/main/java/com/hedera/services/pricing/BaseOperationUsage.java:136]: Got it. Thanks for the clarification!

**Review by Neeharika-Sompalli — APPROVED** [review]
Overall Looks Great !! 👏

**Review by qnswirlds — COMMENTED** [review]
Need to address those code smells

https://sonarcloud.io/project/issues?branch=01735-01722-AddFeeSchUpdPricing&id=com.hedera.hashgraph%3Ahedera-services&resolved=false&types=CODE_SMELL

in the `01722-M-CreateTokenFeeSchUpdOp` branch before merging back to `master`.

**Comment by Neeharika-Sompalli** [comment]
> Need to address those code smells
> 
> https://sonarcloud.io/project/issues?branch=01735-01722-AddFeeSchUpdPricing&id=com.hedera.hashgraph%3Ahedera-services&resolved=false&types=CODE_SMELL
> 
> in the `01722-M-CreateTokenFeeSchUpdOp` branch before merging back to `master`.

@qnswirlds  Will fix these new code smells and open PR

**Pull request merged**